lib/cocoapods/installer/analyzer/target_generator.rb

```diff
@@ -0,0 +1,397 @@
+              dependent_specs.group_by(&:root).each do |root_spec, resolver_specs|
+                all_specs = resolver_specs.map(&:spec)
+                test_specs, specs = all_specs.partition(&:test_specification?)
+                pod_variant = PodVariant.new(specs, test_specs, target_definition.platform, target_definition.uses_frameworks?)
+                hash[root_spec] ||= {}
+                (hash[root_spec][pod_variant] ||= []) << target_definition
+                hash[root_spec].keys.find { |k| k == pod_variant }.test_specs.concat(test_specs).uniq!
+              end
+            end
+
+            pod_targets = distinct_targets.flat_map do |_root, target_definitions_by_variant|
+              suffixes = PodVariantSet.new(target_definitions_by_variant.keys).scope_suffixes
+              target_definitions_by_variant.flat_map do |variant, target_definitions|
+                generate_pod_target(target_definitions, target_inspections, variant.specs + variant.test_specs, sandbox, installation_options, :scope_suffix => suffixes[variant])
+              end
+            end
+          else
+            dedupe_cache = {}
+            pod_targets = resolver_specs_by_target.flat_map do |target_definition, specs|
+              grouped_specs = specs.group_by(&:root).values.uniq
+              grouped_specs.flat_map do |pod_specs|
+                generate_pod_target([target_definition], target_inspections, pod_specs.map(&:spec), sandbox, installation_options).scoped(dedupe_cache)
+              end
+            end
+          end
+
+          all_resolver_specs = resolver_specs_by_target.values.flatten.map(&:spec).uniq
+          pod_targets_by_name = pod_targets.group_by(&:pod_name).each_with_object({}) do |(name, values), hash|
+            # Sort the target by the number of activated subspecs, so that
+            # we prefer a minimal target as transitive dependency.
+            hash[name] = values.sort_by { |pt| pt.specs.count }
+          end
+          pod_targets.each do |target|
+            all_specs = all_resolver_specs.group_by(&:name)
+            dependencies = dependencies_for_specs(target.non_test_specs.to_set, target.platform, all_specs.dup).group_by(&:root)
+            target.dependent_targets = filter_dependencies(dependencies, pod_targets_by_name, target)
+            target.test_dependent_targets_by_spec_name = target.test_specs.each_with_object({}) do |test_spec, hash|
+              test_dependencies = dependencies_for_specs([test_spec], target.platform, all_specs).group_by(&:root)
+              test_dependencies.delete_if { |k| dependencies.key? k }
+              hash[test_spec.name] = filter_dependencies(test_dependencies, pod_targets_by_name, target)
+            end
+          end
+        end
+
+        #-----------------------------------------------------------------------#
+
+        class << self
```

Comment:
please be consistent, and either use `class << self` or `def self.`, but not both

Reply:
I was trying to make this section private class methods, but it seems that private class methods aren't really a thing unless you use `class << self` is that correct? should I use `class << self` for the whole class then?

Reply:
`private_class_ method :foo` is also a thing. I think we should just be consistent within a single class